cogs/GuildSettings.py

```diff
@@ -362,19 +356,33 @@ def get_guild_settings(ctx) -> GuildSettings:
         return GUILD_SETTINGS[str(ctx)]
     return GUILD_SETTINGS[str(get_guild_id(ctx))]
 
-def delete_settings(ctx):
+
+async def delete_settings(ctx):
     global GUILD_SETTINGS
+    guild_id = ''
     if isinstance(ctx, str):
-        del GUILD_SETTINGS[ctx]
+        guild_id = ctx
     elif isinstance(ctx, int):
-        del GUILD_SETTINGS[str(ctx)]
+        guild_id = str(ctx)
     else:
-        del GUILD_SETTINGS[str(get_guild_id(ctx))]
+        guild_id = str(get_guild_id(ctx))
+
+    del GUILD_SETTINGS[guild_id]
+    try:
+        con = await sql.connect('guild_settings.db')
+        await con.execute("PRAGMA foreign_keys = ON")
+        await con.execute("DELETE FROM guild_settings WHERE guild_id = ?", (guild_id,))
+        await con.commit()
+    except sql.Error as e:
+        print(f"Error deleting settings for guild {guild_id}: {e}")
+        return
+    finally:
+        await con.close()
 
 
-def default_settings(ctx) -> GuildSettings:
-    delete_settings(ctx)
-    
+async def default_settings(ctx) -> GuildSettings:

```

Comment:
Need to make sure all calls to this function change (this is a note for myself to check).